Tests/DiligentCoreAPITest/src/ReadOnlyDepthTest.cpp

```diff
@@ -0,0 +1,581 @@
+        Attachments[1].StoreOp      = ATTACHMENT_STORE_OP_STORE;
+
+        SubpassDesc Subpasses[1];
+
+        Subpasses[0].RenderTargetAttachmentCount = 1;
+        AttachmentReference ReadOnlyDepthAttachmentRef{0, RESOURCE_STATE_DEPTH_READ};
+        Subpasses[0].pDepthStencilAttachment = &ReadOnlyDepthAttachmentRef;
+        AttachmentReference RTAttachmentRef{1, RESOURCE_STATE_RENDER_TARGET};
+        Subpasses[0].pRenderTargetAttachments = &RTAttachmentRef;
+
+        RenderPassDesc RPDesc;
+        RPDesc.Name            = "Read only depth test -- color render pass";
+        RPDesc.AttachmentCount = _countof(Attachments);
+        RPDesc.pAttachments    = Attachments;
+        RPDesc.SubpassCount    = _countof(Subpasses);
+        RPDesc.pSubpasses      = Subpasses;
+
+        pDevice->CreateRenderPass(RPDesc, &pRenderPass);
+        VERIFY_EXPR(pRenderPass);
+
+        FramebufferDesc FBDesc;
+        FBDesc.Name               = "Read only depth test -- color framebuffer";
+        FBDesc.pRenderPass        = pRenderPass;
+        FBDesc.AttachmentCount    = _countof(Attachments);
+        ITextureView* pTexViews[] = {m_pReadOnlyDSV, m_pRTV};
+        FBDesc.ppAttachments      = pTexViews;
+
+        pDevice->CreateFramebuffer(FBDesc, &pFramebuffer);
+        VERIFY_EXPR(pFramebuffer);
+    }
+
+    TEXTURE_FORMAT GetDepthFormat() const
+    {
+        const auto& Param = GetParam();
+        return static_cast<TEXTURE_FORMAT>(std::get<0>(Param));
+    }
+
+    static RefCntAutoPtr<IBuffer> sm_pVB;
+
+    RefCntAutoPtr<IPipelineState> m_pDepthPSO;
+    RefCntAutoPtr<IPipelineState> m_pColorPSO;
+
+    RefCntAutoPtr<ITexture>     m_pDepthTexture;
+    RefCntAutoPtr<ITextureView> m_pReadWriteDSV;
+    RefCntAutoPtr<ITextureView> m_pReadOnlyDSV;
+
+    RefCntAutoPtr<ITextureView> m_pRTV;
+
+    RefCntAutoPtr<IShaderResourceBinding> m_pColorSRB;
+
+private:
+    static void InitializeVertexBuffer()
+    {
+        auto* pEnv    = GPUTestingEnvironment::GetInstance();
+        auto* pDevice = pEnv->GetDevice();
+
+        const bool  bOpenGL       = pDevice->GetDeviceInfo().Type == RENDER_DEVICE_TYPE_GL || pDevice->GetDeviceInfo().Type == RENDER_DEVICE_TYPE_GLES;
+        const float GeometryDepth = bOpenGL ? ExpectedDepth * 2.0f - 1.0f : ExpectedDepth;
+
+        // clang-format off
+        float4 VertexData[] =
+        {
+            float4(-1.0f,  -1.0f,  GeometryDepth,  1.f),
+            float4(-1.0f,  +1.0f,  GeometryDepth,  1.f),
+            float4(+1.0f,  +1.0f,  GeometryDepth,  1.f),
+
+            float4(-1.0f,  -1.0f,  GeometryDepth,  1.f),
+            float4(+1.0f,  +1.0f,  GeometryDepth,  1.f),
+            float4(+1.0f,  -1.0f,  GeometryDepth,  1.f)
+        };
+        // clang-format on
+
+        BufferDesc BuffDesc;
+        BuffDesc.Name      = "Test vertex buffer";
+        BuffDesc.BindFlags = BIND_VERTEX_BUFFER;
+        BuffDesc.Size      = sizeof(VertexData);
```

Comment:
For your case, I don't think a vertex buffer is necessary. You can generate it directly in a vertex shader.